src/utils/redirect.ts

```diff
@@ -0,0 +1,12 @@
+import { type NextRequest, NextResponse } from 'next/server';
+import { withLocale } from './i18nPath';
+
+export function redirectWithLocale(
+  req: NextRequest,
+  locale: string,
+  targetPath: string
+) {
+  const url = req.nextUrl.clone();
+  url.pathname = withLocale(locale, targetPath);
```

Comment:
вот здесь тогда будет так:

```suggestion
url.pathname = `/${locale}${targetPath}`;
```